webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -200,7 +200,20 @@ const UploadButton = ({config, repo, reference, path, onDone, onClick, onHide, s
             inProgress: true
         })
         try {
-            await objects.upload(repo.id, reference.id, textRef.current.value, fileRef.current.files[0])
+            if (config.pre_sign_support && config.pre_sign_support_ui) {
+                let getResp = await staging.get(repo.id, reference.id, textRef.current.value, config.pre_sign_support);
+
+                const putResp = await fetch(getResp.presigned_url, {
+                    method: 'PUT',
+                    mode: 'cors',
+                    body: fileRef.current.files[0]
+                });
+
+                let etag = putResp.headers.get('ETag').replace(/["]/g, '')
+                await staging.link(repo.id, reference.id, textRef.current.value, getResp, etag, fileRef.current.files[0].size);
+            } else {
+                await objects.upload(repo.id, reference.id, textRef.current.value, fileRef.current.files[0])
+            }
```

Comment:
Can we have this logic inside the API implementation? It will make the code cleaner

Reply:
It can be done but in involve the staging and the objects which are different API section so I think it's better that way